packages/ember-data/lib/adapters/json-api-adapter.js

```diff
@@ -0,0 +1,80 @@
+/**
+  @module ember-data
+*/
+
+import RESTAdapter from "ember-data/adapters/rest-adapter";
+
+/**
+  @class JSONAPIAdapter
+  @constructor
+  @namespace DS
+  @extends DS.RESTAdapter
+*/
+export default RESTAdapter.extend({
+  defaultSerializer: '-json-api',
+
+  concatenatedProperties: ['headers'],
```

Comment:
If we are gonna do this, should add it to the rest adapter, but it seems like they can also be a CP as per the comments in the rest adapter, so might not work